src/apps/index.ts

```diff
@@ -4,22 +4,63 @@ import i18next from 'i18next';
 import { AppManifest } from '../types';
 
 export const loadApp = (appId: string) => {
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const appManifest = require(`./${appId}/manifest.json`) as AppManifest;
-
-  if (appManifest.translations) {
-    Object.keys(appManifest.translations).forEach((languageKey) => {
-      const translation = appManifest.translations[languageKey];
-      i18next.addResourceBundle(languageKey, `app:${appId}`, translation);
-    });
+  let appManifest: AppManifest | null = null;
+
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    appManifest = require(`./${appId}/manifest.json`) as AppManifest;
+
+    if (appManifest.translations) {
+      Object.keys(appManifest.translations).forEach((languageKey) => {
+        const translation = appManifest?.translations[languageKey];
+        i18next.addResourceBundle(languageKey, `app:${appId}`, translation);
+      });
+    }
+
+    return appManifest;
+  } catch (e) {
+    console.error(`Failed to load app ${appId} - does the ${appId} directory exist in the src/apps directory?`, e);
   }
 
   return appManifest;
-}
+};
 
 export const loadApps = (allowedApps: string[]) => {
- return allowedApps.reduce((apps: Record<string, AppManifest>, appId: string) => {
-    apps[appId] = loadApp(appId);
-    return apps;
-  }, {});
-}
+  // eslint-disable-next-line no-console
+  const loadedApps: Record<string, AppManifest> = {};
+
+  /**
+   * Cycle through the allowedApps array and load the app manifest for each app
+   * This used to be a .reduce function but we may need to do more complicated
+   * work here in the future so we're using a for loop here.
+   */
+  for (let index = 0; index < allowedApps.length; index++) {
+    const appIdentifier = allowedApps[index];
+    
+    if (appIdentifier) {
+      const loadedApp = loadApp(appIdentifier);
+      if (loadedApp) {
+        loadedApps[appIdentifier] = loadedApp;
+      }
+    } else {
+      continue;
+    }
+  }
+
+  /**
+   * Finally, did we have a REACT_APP_PX_DEVELOPMENT_ID environment variable set?
+   * Attempt to load this also.
+   */
+  if (process.env.REACT_APP_PX_DEVELOPMENT_ID) {
```

Comment:
Just one question here, what if this env variable is an empty string or there is a mismatch between the name of the app in the apps folder and the name in the env file? Should this error be handled as well, or it should not create an issue?